internal/engine/upper_test.go

```diff
@@ -96,6 +104,9 @@ func (b *fakeBuildAndDeployer) BuildAndDeploy(ctx context.Context, manifest mode
 		return store.BuildResult{}, err
 	}
 
+	if manifest.IsDockerCompose() {
+		return b.nextBuildResult(imageID), nil
```

Comment:
This is kind of hacky. Any better ideas?

Reply:
yeahhh kind of hacky. fine for now tho -- except that docker compose builds don't return any info in the BuildResult (cuz there's not an image id) -- so maybe change `b.nextBuildResult` to work okay given a nil `imageID`.